sys/net/sock/sock_util.c

```diff
@@ -240,10 +245,20 @@ int sock_tl_str2ep(struct _sock_tl_ep *ep_out, const char *str)
     hostbuf[hostlen] = '\0';
 
     if (!brackets_flag) {
+#if defined(SOCK_HAS_IPV4)
         if (inet_pton(AF_INET, hostbuf, &ep_out->addr.ipv4) == 1) {
             ep_out->family = AF_INET;
             return 0;
         }
+#endif
+#if defined(MODULE_SOCK_DNS)
+        if (IS_ACTIVE(SOCK_HAS_IPV4)) {
+            return sock_dns_query(hostbuf, ep_out->addr.ipv4, AF_INET);
+        }
+        if (IS_ACTIVE(SOCK_HAS_IPV6)) {
+            return sock_dns_query(hostbuf, ep_out->addr.ipv6, AF_INET6);
+        }
```

Comment:
Should `AF_UNSPEC` be considered here as well if the socket supports both IPv4 and IPv6?